packages/components/table/src/table-layout.ts

```diff
@@ -148,18 +148,21 @@ class TableLayout<T> {
       // avoid v-show
       return
     }
+    const { tableLayout } = this.table.props
     this.appendHeight.value = appendWrapper ? appendWrapper.offsetHeight : 0
-    if (this.showHeader && !headerWrapper) return
+    if (this.showHeader && !headerWrapper && tableLayout === 'fixed') {
+      return
+    }
     const headerTrElm: HTMLElement = tableHeader ? tableHeader : null
     const noneHeader = this.headerDisplayNone(headerTrElm)
 
     const headerHeight = (this.headerHeight.value = !this.showHeader
       ? 0
-      : headerWrapper.offsetHeight)
+      : headerWrapper?.offsetHeight || 0)
     if (
       this.showHeader &&
       !noneHeader &&
-      headerWrapper.offsetWidth > 0 &&
+      headerWrapper?.offsetWidth > 0 &&
```

Comment:
**Question:** since the `headerWrapper?.offsetWidth` might be `undefined`, can we do something like:

```ts
const headerWrapperOffsetHeight = headerWrapper?.offsetHeight || 0
// Use this variable for operations
```